radix-engine/src/blueprints/resource/auth_zone.rs

```diff
@@ -291,4 +292,43 @@ impl AuthZoneBlueprint {
 
         Ok(IndexedScryptoValue::from_typed(&proofs))
     }
+
+    pub(crate) fn drop<Y>(
+        input: &IndexedScryptoValue,
+        api: &mut Y,
+    ) -> Result<IndexedScryptoValue, RuntimeError>
+    where
+        Y: KernelSubstateApi<SystemLockData> + ClientApi<RuntimeError>,
+    {
+        // TODO: add `drop` callback for drop atomicity, which will remove the necessity of kernel api.
```

Comment:
I think we used to do this but there were some reasons I moved away from this. I think it was to be able to capture the owned nodes in the dropped object, which a Rust style like `Drop` interface wouldn't really allow.

Reply:
A general direction I'd like to push for is to move drop customisation to clients - they may decide to detach owned objects and let system attempt to auto drop; or they may decide to clean up everything without system assumption.

Even in `rust`, [the `Drop` interface](https://doc.rust-lang.org/std/ops/trait.Drop.html) exposes a `&mut self`, so we can detach owned objects too.